src/plugins/embeddable/server/plugin.ts

```diff
@@ -52,82 +53,20 @@ export class EmbeddableServerPlugin implements Plugin<object, object> {
   }
 
   public start(core: CoreStart) {
-    return {
-      telemetry: this.telemetry,
-      extract: this.extract,
-      inject: this.inject,
+    const commonContract = {
+      getEmbeddableFactory: this.getEmbeddableFactory,
+      getEnhancement: this.getEnhancement,
     };
-  }
-
-  public stop() {}
-
-  private telemetry = (state: EmbeddableInput, telemetryData: Record<string, any> = {}) => {
-    const enhancements: Record<string, any> = state.enhancements || {};
-    const factory = this.getEmbeddableFactory(state.id);
-
-    let telemetry = telemetryBaseEmbeddableInput(state, telemetryData);
-    if (factory) {
-      telemetry = factory.telemetry(state, telemetry);
-    }
-    Object.keys(enhancements).map((key) => {
-      if (!enhancements[key]) return;
-      telemetry = this.getEnhancement(key).telemetry(enhancements[key], telemetry);
-    });
-
-    return telemetry;
-  };
-
-  private extract = (state: EmbeddableInput) => {
-    const enhancements = state.enhancements || {};
-    const factory = this.getEmbeddableFactory(state.id);
-
-    const baseResponse = extractBaseEmbeddableInput(state);
-    let updatedInput = baseResponse.state;
-    const refs = baseResponse.references;
-
-    if (factory) {
-      const factoryResponse = factory.extract(state);
-      updatedInput = factoryResponse.state;
-      refs.push(...factoryResponse.references);
-    }
-
-    updatedInput.enhancements = {};
-    Object.keys(enhancements).forEach((key) => {
-      if (!enhancements[key]) return;
-      const enhancementResult = this.getEnhancement(key).extract(
-        enhancements[key] as SerializableState
-      );
-      refs.push(...enhancementResult.references);
-      updatedInput.enhancements![key] = enhancementResult.state;
-    });
 
     return {
-      state: updatedInput,
-      references: refs,
+      telemetry: getTelemetryFunction(commonContract),
```

Comment:
Could we make this also available in setup phase because we will need this for server side SO migrations 🙏